packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -313,14 +330,11 @@ export class Trail extends Resource {
    * be captured from any one.
    *
    * Be sure to filter the event further down using an event pattern.
+   *
+   * @deprecated - use Trail.onEvent()
```

Comment:
had a similar thought re: stepfunctions - how do we communicate the change of an API to deprecated. It feels like it belongs somewhere in the `CHANGELOG` but I don't know what a good mechanism is.

It's not a breaking change, but worth calling out so users can discover it outside of an IDE.

Reply:
Ideally, the CLI on running `cdk synth` or `cdk deploy` reports a set of warnings for each deprecated API that was used.

Reply:
I think that's a reasonable experience. We do have a `-strict` flag that can fail on warnings for users running in CI. Might be nice if it got a CHANGELOG mention for users who check release notes.